dali/operators/image/crop/bbox_crop.cc

```diff
@@ -617,11 +683,20 @@ class RandomBBoxCropImpl : public OpImplBase<CPUBackend> {
 
   bool ValidOverlap(const Box<ndim, float> &crop,
                     span<const Box<ndim, float>> boxes,
-                    float threshold) {
-    return std::all_of(boxes.begin(), boxes.end(),
-      [&crop, threshold](const Box<ndim, float> &box) {
-        return intersection_over_union(crop, box) >= threshold;
-      });
+                    float threshold, ThresholdType threshold_type) {
+    std::function<bool(const Box<ndim, float> &box)> f;
+    if (threshold_type_ == ThresholdType::Overlap) {
+      f = [&crop, threshold](const Box<ndim, float> &box) {
+          float overlap = volume(intersection(crop, box)) / static_cast<float>(volume(box));
+          return overlap >= threshold;
+        };
+    } else {  // IoU
+      f = [&crop, threshold](const Box<ndim, float> &box) {
+          return intersection_over_union(crop, box) >= threshold;
+        };
+    }
+
+    return std::any_of(boxes.begin(), boxes.end(), f);
```

Comment:
Why has it changed from `all_of` to `any_of`?

Reply:
I'll revert to `all_of` as default, but provide an option to accept the crop if at least one box overlaps sufficiently